core/arch/arm/kernel/elf_load.c

```diff
@@ -471,6 +472,25 @@ TEE_Result elf_resolve_symbol(struct elf_load_state *state,
 	return TEE_ERROR_ITEM_NOT_FOUND;
 }
 
+static size_t name_idx_nospec(Elf32_Sym *sym_tab, size_t num_syms, size_t sym_idx)
+{
+	Elf32_Word *lower = (Elf32_Word *)sym_tab ;
```

Comment:
Why this casting to `Elf32_Word` and the explicit dependency on the layout of `Elf32_Sym`?
Wouldn't `&sym_tab->st_name` work for this line and the same pattern for the others?

Reply:
Yes, code updated.